### PR TITLE
Authorization - resource section

### LINKED ISSUE #396: Authorization implemented - resource section
Description of roles in perun: https://is.muni.cz/auth/th/t4sdm/Master_Thesis_final.pdf
Where to find policies what needs to be used: perun-roles.yml in perun repository(backend)
What needs to be done:
In entity-overview hide items that user dont have role to see them - what does that means - eg. - vo - application item - user shouldnt have see that item if he doesnt have the authority to call the method getApplication(example)
Same rules are applicated to side-menu.
Also check the entity description(there could be link into the user detail or facility detail) and you must check if that link should be available.

Then when we go to the eg. vo-groups page we agreed on:
1. button - create/add - this should be easy look on vo-groups button add, I already did an example how to do that
2. button - remove/delete - when checking/unchecking the select checkbox in the table always check if the user have the authority to delete this group and if dont make the delete button disabled and make the tooltip on that button say that he doesnt have the priviledge to delete some of the selected groups.
3. clicking on the detail of the entity - always check on first entity in the list if the user have priviledge to go into the detail of entity. There is example in vo-groups.

Your fake account that you have created use for the testing if your solution works. If that wouldnt work there would be PerunPriviledge exception notification showing up.

### PR DESCRIPTION
**Pull request by xkostka2**
* Authorization for resource section was added (except for managers page)
* Resolves #396

### CODE REVIEW

**Review by xkureck — CHANGES_REQUESTED** [review]
image
Some bug occured that assigned services are accesible only from facility part, if vo-manager or resource-manager can get to this part of application the page should be accesible from vo part too.
image
There should be some tooltip why the attributes cannot be checked(selected)
image
(tested as a group admin) Missing authorization to edit resource button.
image
Missing authorization to link to the facility

**Review by bodnara — CHANGES_REQUESTED** [review]
image
As resource admin I get error when accessing Assigned services.
I believe this is caused by missing authorization for overview buttons as well as sidemenu items.
image
I would also suggest hiding checkbox column if there is no purpose for them

**Review by xkureck — CHANGES_REQUESTED** [review]
image
Access to the services should only have perun admin.

File: apps/admin-gui/src/app/shared/side-menu/side-menu-item.service.ts [hunk 1]
```diff
@@ -322,6 +289,48 @@ export class SideMenuItemService {
...
+        url: [baseUrl, 'groups'],
+        activatedRegex: `${regexStart}/\\d+/resources/\\d+/groups$`
+      })
+    }
+    if(this.guiAuthResolver.isPerunAdmin()){
+      links.push({
+        label: 'MENU_ITEMS.RESOURCE.ASSIGNED_SERVICES',
+        url: [baseUrl, 'services'],
+        activatedRegex: `${regexStart}/\\d+/resources/\\d+/services$`
+      })
```
Comment (xkureck) [inline apps/admin-gui/src/app/shared/side-menu/side-menu-item.service.ts:310]: We made some misscomunication, to get into the assigned services and see them/add/remove them should be done according to the policies.

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html [hunk 2]
```diff
@@ -18,5 +18,7 @@ <h1 class="page-subtitle">{{'RESOURCE_DETAIL.ASSIGNED_SERVICES.TITLE' | translat
   [services]="assignedServices"
   [selection]="selected"
   [filterValue]="filterValue"
+  [hideColumns]="removeServiceAuth ? []: ['select']"
+  [disableRouting]="!guiAuthResolver.isAuthorized('getServiceById_int_policy',[])"
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-services/resource-assigned-services.component.html:22]: This routing to the service detail should be enabled only for perun admins

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-groups/resource-groups.component.html [hunk 3]
```diff
@@ -1,10 +1,10 @@
 <h1 class="page-subtitle">{{'RESOURCE_DETAIL.ASSIGNED_GROUPS.TITLE' | translate}}</h1>
 <perun-web-apps-refresh-button (refresh)="loadAllGroups()"></perun-web-apps-refresh-button>
-<button mat-flat-button color="accent" (click)="addGroup()">
+<button (click)="addGroup()" *ngIf="!loadingResource && guiAuthResolver.isAuthorized('assignGroupsToResource_List<Group>_Resource_policy',[resource])" color="accent" mat-flat-button>
   {{'RESOURCE_DETAIL.ASSIGNED_GROUPS.ADD_GROUP' | translate}}
 </button>
-<button mat-flat-button color="warn" class="ml-2" [disabled]="selected.selected.length === 0" (click)="removeGroups()">
-  {{'RESOURCE_DETAIL.ASSIGNED_GROUPS.REMOVE_GROUP' | translate}}
+<button (click)="removeGroups()" [disabled]="selected.selected.length === 0 || !canRemoveGroups()" class="ml-2 mr-2" color="warn" mat-flat-button>
+  <span  [matTooltipDisabled]="canRemoveGroups()" matTooltip="{{'RESOURCE_DETAIL.ASSIGNED_GROUPS.REMOVE_TOOLTIP' | translate}}">{{'RESOURCE_DETAIL.ASSIGNED_GROUPS.REMOVE_GROUP' | translate}}</span>
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-groups/resource-groups.component.html:7]: at least split this code into the - one "thing" one row, this is very hard to read

File: apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-overview/resource-overview.component.ts [hunk 4]
```diff
@@ -40,25 +42,28 @@ export class ResourceOverviewComponent implements OnInit {
...
+      this.navItems.push({
         cssIcon: 'perun-group',
         url: `${urlStart}/resources/${this.resource.id}/groups`,
         label: 'MENU_ITEMS.RESOURCE.ASSIGNED_GROUPS',
         style: 'resource-btn'
-      },
-      {
+      })
+    }
+    if(this.guiAuthResolver.isPerunAdmin()){
```
Comment (xkureck) [inline apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-overview/resource-overview.component.ts:54]: also there we made a miscomunication

File: libs/perun/components/src/lib/attributes-list/attribute-value/attribute-value-map/attribute-value-map.component.html [hunk 5]
```diff
@@ -10,7 +10,8 @@
                 matInput
                 required
                 type="text"
-                (click)="readonly && attribute.displayName !== undefined && showValue(keys[i], attribute.displayName)"
+                class="overflow-ellipsis"
+                (click)="readonly && showValue(keys[i], attribute.displayName)"
```
Comment (xkureck) [inline libs/perun/components/src/lib/attributes-list/attribute-value/attribute-value-map/attribute-value-map.component.html:14]: previous code: `(click)="readonly && attribute.displayName !== undefined && showValue(keys[i], attribute.displayName)"
` wasnt the attribute.displayName !== undefined important?

**Review by xkureck — CHANGES_REQUESTED** [review]
image
You rebased the adam PR that is merged but now there is two tooltips for checkbox in attributes. Please keep the Adams way of doing the tooltips.

**Review by xkureck — APPROVED** [review]

**Pull request merged**